### PR TITLE
MF-174 - Add logger interface and go-kit logger implementation

### LINKED ISSUE #174: Wrap the usage of external logger in our own implementation
Instead of directly using a logger provided by `go-kit`, implement Mainflux logger that depends on it (technically it can depend on any other 3rd party logger). 

**Acceptance criteria:**
- all logs are JSON formatted and emitted to stdout
- all logs must contain an UTC timestamp field named `ts` as ISO-8601 string
- logger should expose 3 methods: `Info`, `Warn`, `Error` (each of these will set `level` field)
- ensure that all services are using only our implementation

### PR DESCRIPTION
**Pull request by anovakovic01**
This pull request resolves #174.

**Summary:**
- Add wrapper logger interface and wrap go-kit logger with it
- Add tests for info, warn and error methods

### CODE REVIEW

**Review by drasko — DISMISSED** [review]
LGTM

File: log/kitlog/logger.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+// New returns wrapped go kit logger.
+func New(out io.Writer) log.Logger {
+	l := logkit.NewJSONLogger(logkit.NewSyncWriter(out))
+	l = logkit.With(l, "ts", logkit.DefaultTimestampUTC)
+	return &logger{log.All, l}
+}
+
+func (l *logger) SetLevel(lvl log.Level) {
```
Comment (mijicd) [inline log/kitlog/logger.go]: Please remove this.

File: log/kitlog/logger.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	l.lvl = lvl
+}
+
+func (l logger) Info(obj interface{}) {
```
Comment (mijicd) [inline log/kitlog/logger.go]: Logging methods should accept only the string parameter.

File: log/kitlog/logger.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+func (l logger) Warn(obj interface{}) {
+	l.Log(log.Warn, obj)
+}
+
+func (l logger) Error(obj interface{}) {
+	l.Log(log.Error, obj)
+}
+
+func (l logger) Log(lvl log.Level, obj interface{}) {
```
Comment (mijicd) [inline log/kitlog/logger.go]: Please remove this.

File: log/level.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package log
+
+const (
+	// All level is used when logging everything.
+	All Level = iota
```
Comment (mijicd) [inline log/level.go]: This won't exist.

File: log/level.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+var levels = map[Level]string{
+	Error: "error",
+	Warn:  "warn",
+	Info:  "info",
+}
+
+// AddLevel adds custom level's name to levels map.
+func AddLevel(lvl Level, name string) {
```
Comment (mijicd) [inline log/level.go]: Please remove this.

File: log/logger.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package log
+
+// Logger specifies logging API.
+type Logger interface {
```
Comment (mijicd) [inline log/logger.go]: See my remarks given above. Also, rename the package to `logger`.

**Review by mijicd — CHANGES_REQUESTED** [review]
Besides the explicitly requested changes, make sure that all logging usages are updated to use the new logger.

**Review by mijicd — CHANGES_REQUESTED** [review]
In case that I missed something, use `warn` for all erroneous cases that does not "break" the service, e.g. failed normalization can happen for non-SenML messages. Use `error` whenever there is a infrastructure/service failure, e.g. database or NATS connection failure.

File: logger/logger_test.go [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+	"fmt"
+	"testing"
+
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInfo(t *testing.T) {
+	cases := map[string]struct {
+		input string
```
Comment (mijicd) [inline logger/logger_test.go]: The test is missing the expected output. I would recommend to restructure it in such a way that is easy to acquire the contents and verify it, i.e. provide an implementation of `io.Writer` for test purposes.

**Review by mijicd — CHANGES_REQUESTED** [review]
Getting there.

**Review by mijicd — APPROVED** [review]

**Pull request merged**